apps/music/js/metadata/ogg.js

```diff
@@ -172,16 +239,70 @@ var OggMetadata = (function() {
    */
   function readComment(page) {
     if (page.remaining() < 4) { // 4 bytes for comment-length variable
-      // TODO: handle metadata that uses multiple pages
-      throw new EndOfPageError();
+      throw new EndOfPageError({
+        page: page,
+        index: page.tell(),
+        size: page.remaining()
+      });
     }
     var comment_length = page.readUnsignedInt(true);
     if (comment_length > page.remaining()) {
-      // TODO: handle metadata that uses multiple pages
-      throw new EndOfPageError();
+      throw new EndOfPageError({
+        page: page,
+        index: page.tell(),
+        size: page.remaining()
+      }, comment_length);
     }
 
     var comment = page.readUTF8Text(comment_length);
+    return decodeComment(comment);
+  }
+
+  function readContinuedComment(leftovers, comment_length, page) {
+    var buffer;
+    var left;
+    if (comment_length === undefined) {
+      buffer = new Uint8Array(4);
+      var size = Math.min(leftovers.size, 4);
+      var val = leftovers.page.readUnsignedByteArray(size);
+      buffer.set(val, 0);
+      if (size < 4) {
+        left = 4 - size;
+        val = page.readUnsginedByteArray(left);
+        buffer.set(val, size);
+      }
+      comment_length = DataView(buffer.buffer).getUint32(0, true);
+    }
+
+    left = leftovers.page.remaining();
+    var newBuffer = new Uint8Array(left + page.remaining());
+    if (left > 0) {
+      buffer = leftovers.page.readUnsignedByteArray(left);
+      newBuffer.set(buffer, 0);
+    }
+    buffer = page.readUnsignedByteArray(Math.min(comment_length - left,
+                                                 page.remaining()));
+    newBuffer.set(buffer, left);
+
+    var view = BlobView.getFromArrayBuffer(newBuffer.buffer, 0,
+                                           newBuffer.byteLength,
+                                           true);
+    if (comment_length > view.remaining()) {
+      throw new EndOfPageError({
+        page: view,
+        index: 0,
+        size: newBuffer.byteLength
+      }, comment_length);
+    }
```

Comment:
This code seems awfully complex for what's going on. I think I'd prefer something that works like BlobView, but is smart enough to expand its buffer size until it fits the whole field. Then this function wouldn't even be necessary, and we could just use `readComment`. Essentially, `readComment`'s job would be to do its best to read a whole comment, but if it can't, it would indicate failure, and ask the caller to expand the buffer and try again.